packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,220 @@
+# Checkbox Decisions <!-- omit in toc -->
+
+- [Design](#design)
+- [Anatomy](#anatomy)
+- [`Checkbox` API](#checkbox-api)
+  - [Examples:](#examples)
+    - [Basic](#basic)
+    - [Validations](#validations)
+- [`CheckboxGroup` API](#checkboxgroup-api)
+  - [Examples:](#examples-1)
+    - [Basic](#basic-1)
+    - [Controlled, Uncontrolled](#controlled-uncontrolled)
+  - [Note on CheckboxGroup](#note-on-checkboxgroup)
+- [Component Architecture](#component-architecture)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=13227%3A163026) to all variants of the Checkbox component
+
+## `Checkbox` API
+
+| Prop                | Type       | Required | Default   | Description                                                                                                                                              |
+| ------------------- | ---------- | -------- | --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| children            | `string`   | No       | undefined | The text to be rendered as CheckboxLabel                                                                                                                 |
+| helpText          | `string`   | No       | undefined | The helper text to be rendered                                                                                                                           |
+| isChecked           | `boolean`  | No       | false     | state of the checkbox, if explicitly set the checkbox will be controlled                                                                                 |
+| defaultChecked      | `boolean`  | No       | false     | default state of the checkbox, if isChecked is not provided the checkbox will be uncontrolled                                                            |
+| onChange            | `Function` | No       | undefined | The function to be called when the checkbox state changes.                                                                                               |
+| isIndeterminate     | `boolean`  | No       | false     | mixed state of the checkbox                                                                                                                              |
```

Comment:
In which cases can there be an indeterminate state? Let's add in description too

Reply:
Will do, isIndeterminate will be an independent state boolean. which the user will handle the logic. 

[Live demo: ](https://61c19ee8d3d282003ac1d81c-evllbcmcsj.chromatic.com/?path=/story/components-checkbox--indeterminate-checkbox&globals=measureEnabled:false)